File: ReverseInteger/ReverseInteger.c
#include <math.h>

int reverse(int x) {
    int reverse = 0;
    int i = x;
    int P = 0;
    while (i != 0) {
        i = i / 10;
        P++;
    };
    while (x != 0) {
        int y = x % 10;
        if (y * pow(10, (P - 1)) < -2147483648 || y * pow(10, (P - 1)) > 2147483647) {
            return 0;
        };
        if (reverse + y * pow(10, (P - 1)) < -2147483648 || reverse + y * pow(10, (P - 1)) > 2147483647) {
            return 0;
        };
        reverse = reverse + y * ((int)pow(10, (P--) - 1));
        x = x / 10;
    };
    return reverse;
}

File: ReverseInteger/ReverseInteger.py
class Solution:
    def reverse(self, x: int) -> int:
        if x == 0:
            return 0
        sign = -1 if x < 0 else 1
        if sign == -1:
            x *= -1
        digit = []
        while x > 0:
            d = x % 10
            digit.append(str(d))
            x = x // 10
        num = int("".join([d for d in digit])) * sign
        if num > 2147483647 or num < -2147483648:
            return 0
        return num
